cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/token/ConsistencyLevel.java

```diff
@@ -82,19 +119,25 @@ public boolean checkConsistency(Collection<? extends CassandraInstance> failedIn
         },
         QUORUM
         {
+            // Keyspaces exist with RF 1 or 2
             @Override
             public boolean isLocal()
             {
                 return false;
             }
 
             @Override
-            public boolean checkConsistency(Collection<? extends CassandraInstance> failedInsts,
-                                            ReplicationFactor replicationFactor,
-                                            String localDC)
+            public boolean checkConsistency(final CassandraRing ring,
+                                            final TokenRangeMapping<? extends CassandraInstance> tokenRangeMapping,
+                                            final Collection<? extends CassandraInstance> failedInsts,
+                                            final String localDC)
             {
-                int rf = replicationFactor.getTotalReplicationFactor();
-                return failedInsts.size() <= (rf - (rf / 2 + 1));
+                Set<String> replacingInstances = tokenRangeMapping.getReplacementInstances();
+                Set<String> failedInstanceIPs = failedInsts.stream().map(CassandraInstance::getIpAddress).collect(Collectors.toSet());
+                final long writeReplicaCount = maybeUpdateWriteReplicasForReplacements(tokenRangeMapping.getWriteReplicas(),
+                                                                                       replacingInstances,
+                                                                                       failedInstanceIPs);
```

Comment:
Similar to the API change comment. It is not the consistency level variants' responsibility to calculate the effective write replica set.

Reply:
Addressed